doc/development_process/api_lifecycle.rst

```diff
@@ -77,16 +77,92 @@ An API can be declared ``stable`` after fulfilling the following requirements:
 - Stable APIs can get backward compatible updates, bug fixes and security fixes
   at any time.
 
+.. _stable_api_changes:
+
+Introducing incompatible changes
+================================
+
+A stable API, as described above strives to remain backwards-compatible through
+its life-cycle. There are however cases where fulfilling this objective prevents
+technical progress or is simply unfeasible without unreasonable burden on the
+maintenance of the API and its implementation(s).
+
+In order to restrict and control the introduction of a change that breaks the
+promise of backwards compatibility the following steps must be followed whenever
+such a change is considered necessary in order to accept it in the project:
+
+#. An :ref:`RFC issue <rfcs>` must be opened on GitHub with the following
+   content:
+
+   .. code-block:: none
+
+      Title:     RFC: API Change: <subsystem>
+      Contents:  - Problem Description:
+                   - Background information on why the change is required
+                 - Proposed Change (detailed):
+                   - Brief description of the API change
+                 - Detailed RFC:
+                   - Function call changes
+                   - Device Tree changes (source and bindings)
+                   - Kconfig option changes
+                 - Dependencies:
+                   - Impact to users of the API, including the steps required
+                     to adapt out-of-tree users of the API to the change
+
+   Instead of a written description of the changes, the RFC issue may link to a
+   Pull Request containing those changes in code form.
+#. The RFC issue must be labeled with the GitHub ``Stable API Change`` label
+#. The RFC issue must be submitted for discussion in the next `Zephyr API meeting`_
+#. An email must be sent to the ``devel`` mailing list with a subject identical
+   to the RFC issue title and that links to the RFC issue
+
+The RFC will then receive feedback through issue comments and will also be
+discussed in the Zephyr API meeting, where the stakeholders and the community at
+large will have a chance to discuss it in detail.
+
+Finally, and if not done as part of the first step, a Pull Request must be
+opened on GitHub. It is left to the person proposing the change to decide
+whether to introduce both the RFC and the Pull Request at the same time or to
+wait until the RFC has gathered consensus enough so that the implementation can
+proceed with confidence that it will be accepted.
+The Pull Request must include the following:
+
+- A title that matches the RFC issue
+- A link to the RFC issue
+- The actual changes to the API
+
+  - Changes to the API header file
+  - Changes to the API implementation(s)
+  - Changes to the relevant API documentation
+  - Changes to Device Tree source and bindings
+
+- The changes required to adapt all in-tree users of the API to the change
```

Comment:
If the stable API changes refer to driver API are we asking the author of the PR to single-handedly update all the existing drivers and in-tree users of the API?

Reply:
No, you are right. I can amend that